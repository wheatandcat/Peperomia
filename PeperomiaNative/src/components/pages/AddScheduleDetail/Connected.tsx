import { SQLite } from "expo-sqlite";
import React, { Component } from "react";
import { NavigationScreenProp, NavigationRoute } from "react-navigation";
import uuidv1 from "uuid/v1";
import { db } from "../../../lib/db";
import {
  ItemDetailParam,
  insert as insertItemDetail,
  ItemDetail
} from "../../../lib/db/itemDetail";
import getKind from "../../../lib/getKind";
import { SuggestItem } from "../../../lib/suggest";
import { Consumer as ItemsConsumer } from "../../../containers/Items";
import Page from "../../templates/CreateScheduleDetail/Page";

export interface State extends ItemDetailParam {
  iconSelected: boolean;
  suggestList: SuggestItem[];
}

interface Props extends ItemDetailParam {
  navigation: NavigationScreenProp<NavigationRoute>;
}

interface PlanProps extends Props {
  itemDetails: ItemDetail[];
  refreshData: () => void;
}

export default class extends Component<Props> {
  render() {
    return (
      <ItemsConsumer>
        {({ refreshData, itemDetails }: any) => (
          <Plan
            {...this.props}
            refreshData={refreshData}
            itemDetails={itemDetails}
          />
        )}
      </ItemsConsumer>
    );
  }
}

class Plan extends Component<PlanProps, State> {
  state = {
    title: this.props.title || "",
    place: this.props.place || "",
    url: this.props.url || "",
    memo: this.props.memo || "",
    moveMinutes: this.props.moveMinutes || 0,
    kind: this.props.kind,
    priority: this.props.priority,
    iconSelected: false,
    suggestList: []
  };

  componentDidMount() {
    const suggestList = this.props.itemDetails.map(itemDetail => ({
      title: itemDetail.title,
      kind: itemDetail.kind
    }));

    this.setState({
      suggestList
    });
  }

  componentDidUpdate() {
    const kind = this.props.navigation.getParam("kind", "");

    if (!kind) {
      return;
    }

    if (this.state.kind !== kind) {
      this.setState({ kind, iconSelected: true });
    }
  }

  onDismiss = () => {
    this.props.navigation.goBack();
  };

  onSave = (
    title: string,
    kind: string,
    place: string,
    url: string,
    memo: string,
    time: number
  ) => {
    const itemId = this.props.navigation.getParam("itemId", "1");
    const priority = this.props.navigation.getParam("priority", "1");

    db.transaction((tx: SQLite.Transaction) => {
      const itemDetail: ItemDetail = {
        itemId,
        title,
        place,
        url,
        memo,
        kind,
        moveMinutes: time,
        priority: Number(priority)
      };

      insertItemDetail(tx, itemDetail, this.save);
    });
  };

  save = () => {
    const itemId = this.props.navigation.getParam("itemId", "1");

    this.props.navigation.navigate("Schedule", {
      itemId,
      refresh: uuidv1()
    });

    this.props.refreshData();
  };

  onIcons = (title: string) => {
    this.props.navigation.navigate("Icons", {
      kind: getKind(title),
      onSelectIcon: (kind: string) => {
        this.props.navigation.navigate("AddScheduleDetail", {
          kind: kind
        });
      },
      onDismiss: () => {
        this.props.navigation.navigate("AddScheduleDetail");
      },
      photo: false
    });
  };

  render() {
    return (
      <Page
        title={this.state.title}
        kind={this.state.kind}
        place={this.state.place}
        url={this.state.url}
        memo={this.state.memo}
        time={this.state.moveMinutes}
        suggestList={this.state.suggestList}
        iconSelected={this.state.iconSelected}
        onDismiss={this.onDismiss}
        onSave={this.onSave}
        onIcons={this.onIcons}
      />
    );
  }
}
