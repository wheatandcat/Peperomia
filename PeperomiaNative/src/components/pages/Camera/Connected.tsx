import React, { Component } from 'react';
import { NavigationScreenProp, NavigationRoute } from 'react-navigation';
import EStyleSheet from 'react-native-extended-stylesheet';
import { TouchableOpacity, View } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import Page, { Props as PageProps } from './Page';

interface Props extends PageProps {
  navigation: NavigationScreenProp<NavigationRoute>;
}

export default class extends Component<Props> {
  static navigationOptions = ({
    navigation,
  }: {
    navigation: NavigationScreenProp<NavigationRoute>;
  }) => {
    const { params = {} } = navigation.state;

    return {
      title: 'アイコン写真を取る',
      headerLeft: (
        <View style={styles.headerLeft}>
          <TouchableOpacity onPress={params.onDismiss}>
            <MaterialCommunityIcons name="close" size={25} />
          </TouchableOpacity>
        </View>
      ),
    };
  };

  componentDidMount() {
    const onDismiss = this.props.navigation.getParam('onDismiss', () => {});

    this.props.navigation.setParams({
      onDismiss: onDismiss,
    });
  }

  onPicture = (image?: string) => {
    const onPicture = this.props.navigation.getParam('onPicture', () => {});

    onPicture(image);
  };

  render() {
    return <Page onPicture={this.onPicture} />;
  }
}

const styles = EStyleSheet.create({
  headerLeft: {
    left: 10,
  },
});
