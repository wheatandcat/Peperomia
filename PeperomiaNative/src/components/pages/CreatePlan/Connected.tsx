import React, { useState, memo, useEffect, useCallback } from 'react';
import { Alert } from 'react-native';
import { ContextProps as ItemContextProps } from 'containers/Items';
import { ContextProps as AuthContextProps } from 'containers/Auth';
import { SuggestItem } from 'lib/suggest';
import getKind from 'lib/getKind';
import { useDidMount } from 'hooks/index';
import { createItem } from 'lib/item';
import { createCalendar } from 'lib/calendar';
import Page from 'components/templates/CreatePlan/Page';
import { Props } from './';

type State = {
  input: {
    title: string;
    date: string;
  };
  kind: string;
  suggestList: SuggestItem[];
};

export type ConnectedProps = Props &
  Pick<ItemContextProps, 'items' | 'refreshData' | 'calendars'> &
  Pick<AuthContextProps, 'uid'>;

const Connected: React.FC<ConnectedProps> = memo((props) => {
  const [state, setState] = useState<State>({
    input: { title: '', date: '' },
    kind: '',
    suggestList: [],
  });

  const date = props.route?.params?.date || '';
  const kind = props.route?.params?.kind || '';

  useDidMount(() => {
    const suggestList = (props.items || []).map((item) => ({
      title: item.title,
      kind: item.kind,
    }));

    setState((s) => ({
      ...s,
      suggestList,
      input: {
        ...s.input,
        date,
      },
    }));
  });

  useEffect(() => {
    if (kind && kind !== state.kind) {
      setState((s) => ({
        ...s,
        kind,
      }));
    }
  }, [kind, state.kind]);

  const onIcons = useCallback(() => {
    props.navigation.navigate('Icons', {
      kind: getKind(state.input.title),
      onSelectIcon: (selectedKind: string) => {
        props.navigation.navigate('CreatePlan', {
          kind: selectedKind,
        });
      },
      onDismiss: () => {
        props.navigation.navigate('CreatePlan');
      },
      photo: true,
    });
  }, [props.navigation, state.input.title]);

  const pushCreateSchedule = useCallback(
    (itemId: number | string) => {
      if (props.refreshData) {
        props.refreshData();
      }

      props.navigation.navigate('CreateSchedule', {
        itemId,
        title: state.input.title,
      });
    },
    [props, state.input.title]
  );

  const onInput = useCallback((name: string, value: string) => {
    setState((s) => ({
      ...s,
      input: {
        ...s.input,
        [name]: value,
      },
    }));
  }, []);

  const save = useCallback(
    async (insertId: number | string) => {
      if (state.input.date) {
        // 日付のデータがある場合ははcalendarに登録する
        const calendar = {
          itemId: insertId,
          date: state.input.date,
        };

        const insertID = await createCalendar(props.uid, calendar);

        if (!insertID) {
          Alert.alert('保存に失敗しました');
          return;
        }

        pushCreateSchedule(insertId);
      } else {
        pushCreateSchedule(insertId);
      }
    },
    [pushCreateSchedule, state.input.date, props.uid]
  );

  const onSave = useCallback(async () => {
    if (state.input.date) {
      const check = (props.calendars || []).find(
        (calendar) => calendar.date === state.input.date
      );

      if (check) {
        Alert.alert('同じ日にスケジュールが既に登録されています');
        return;
      }
    }

    const item = {
      title: state.input.title,
      kind: state.kind || getKind(state.input.title),
    };

    const insertID = await createItem(props.uid, item);
    if (!insertID) {
      Alert.alert('保存に失敗しました');
      return;
    }

    save(insertID);
  }, [
    props.calendars,
    save,
    state.input.date,
    state.input.title,
    state.kind,
    props.uid,
  ]);

  const onHome = useCallback(() => {
    props.navigation.goBack();
  }, [props.navigation]);

  return (
    <Page
      mode="new"
      title={state.input.title}
      date={state.input.date}
      kind={state.kind}
      suggestList={state.suggestList}
      onInput={onInput}
      onSave={onSave}
      onIcons={onIcons}
      onHome={onHome}
    />
  );
});

export default Connected;
