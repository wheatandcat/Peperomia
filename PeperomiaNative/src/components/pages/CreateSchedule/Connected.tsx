import React, { useState, memo, useEffect, useCallback } from 'react';
import { NavigationScreenProp, NavigationRoute } from 'react-navigation';
import { Alert } from 'react-native';
import { Item } from '../../../lib/db/item';
import { ItemDetail } from '../../../lib/db/itemDetail';
import { useDidMount } from '../../../hooks/index';
import { getItemByID } from '../../../lib/item';
import { getItemDetails, updateItemDetail } from '../../../lib/itemDetail';
import Page from '../../templates/CreateSchedule/Page';

type Props = {
  navigation: NavigationScreenProp<NavigationRoute>;
};

type State = {
  item: Item;
  itemDetails: ItemDetail[];
  refresh: string;
};

export default memo((props: Props) => {
  const [state, setState] = useState<State>({
    item: { title: '', kind: '' },
    itemDetails: [],
    refresh: '0',
  });

  const setItemDetails = useCallback(
    (data: ItemDetail[]) => {
      const prioritys = data.map(item => item.priority);
      const uniquePrioritys = prioritys.filter(
        (x: number, i: number, self: number[]) => self.indexOf(x) === i
      );

      // priorityが重複していない
      if (prioritys.length === uniquePrioritys.length) {
        props.navigation.setParams({
          itemDetails: data,
        });
        setState(s => ({
          ...s,
          itemDetails: data,
        }));
      }

      // priorityが重複している場合はid順でpriorityをupdateする
      const itemDetails: ItemDetail[] = data.map(
        (itemDetail: ItemDetail, index: number) => ({
          ...itemDetail,
          priority: index + 1,
        })
      );

      itemDetails.forEach(async (itemDetail, index) => {
        const v = {
          ...itemDetail,
          id: itemDetail.id || '',
          priority: index + 1,
        };

        const ok = await updateItemDetail(null, v);
        if (!ok) {
          Alert.alert('保存に失敗しました');
          return;
        }
      });

      setState(s => ({
        ...s,
        itemDetails: itemDetails,
      }));
    },
    [props.navigation]
  );

  const setItem = useCallback((data: Item) => {
    setState(s => ({
      ...s,
      item: {
        title: data.title,
        kind: data.kind,
      },
    }));
  }, []);

  useDidMount(() => {
    const itemId = props.navigation.getParam('itemId', '1');

    const setItemByItemID = async () => {
      const item = await getItemByID<Item>(null, itemId);
      setItem(item);
    };

    const setItemDetailsByItemID = async () => {
      const itemDetails = await getItemDetails<ItemDetail[]>(null, itemId);
      setItemDetails(itemDetails);
    };

    setItemByItemID();
    setItemDetailsByItemID();
  });

  useEffect(() => {
    const refresh = props.navigation.getParam('refresh', '0');

    const setItemDetailsByItemID = async (itemId: string) => {
      const itemDetails = await getItemDetails<ItemDetail[]>(null, itemId);
      setItemDetails(itemDetails);
    };

    if (refresh !== state.refresh) {
      const itemId = props.navigation.getParam('itemId', '1');
      setItemDetailsByItemID(itemId).then(() => {
        setState(s => ({ ...s, refresh }));
      });
    }
  }, [props.navigation, setItemDetails, state.refresh]);

  const onCreateScheduleDetail = useCallback(() => {
    const itemId = props.navigation.getParam('itemId', '1');

    props.navigation.navigate('CreateScheduleDetail', {
      itemId,
    });
  }, [props.navigation]);

  const onScheduleDetail = useCallback(
    (id: string) => {
      props.navigation.navigate('ScheduleDetail', {
        scheduleDetailId: id,
        priority: state.itemDetails.length + 1,
      });
    },
    [props.navigation, state.itemDetails.length]
  );

  const onFinish = useCallback(() => {
    if (state.itemDetails.length === 0) {
      Alert.alert(
        'まだ予定の設定がありません',
        '本当に完了しますか？',
        [
          {
            text: 'キャンセル',
            style: 'cancel',
          },
          {
            text: '完了する',
            onPress: () => {
              props.navigation.navigate('Home', { refresh: true });
            },
          },
        ],
        { cancelable: false }
      );
    } else {
      props.navigation.navigate('Home', { refresh: true });
    }
  }, [props.navigation, state.itemDetails.length]);

  const onGoBack = useCallback(() => {
    if (state.itemDetails.length === 0) {
      Alert.alert(
        'まだ予定の設定がありません',
        '本当に閉じますか？',
        [
          {
            text: 'キャンセル',
            style: 'cancel',
          },
          {
            text: '閉じる',
            onPress: () => {
              props.navigation.navigate('Home', { refresh: true });
            },
          },
        ],
        { cancelable: false }
      );
    } else {
      props.navigation.navigate('Home', { refresh: true });
    }
  }, [props.navigation, state.itemDetails.length]);

  return (
    <Page
      title={state.item.title}
      kind={state.item.kind}
      data={state.itemDetails}
      onScheduleDetail={onScheduleDetail}
      onCreateScheduleDetail={onCreateScheduleDetail}
      onFinish={onFinish}
      onGoBack={onGoBack}
    />
  );
});
