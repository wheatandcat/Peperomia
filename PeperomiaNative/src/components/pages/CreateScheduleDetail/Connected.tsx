import * as SQLite from "expo-sqlite";
import React, { Component } from "react";
import { NavigationScreenProp, NavigationRoute } from "react-navigation";
import uuidv1 from "uuid/v1";
import { db, ResultError } from "../../../lib/db";
import {
  ItemDetailParam,
  insert as insertItemDetail,
  countByItemId,
  ItemDetail
} from "../../../lib/db/itemDetail";
import { SuggestItem } from "../../../lib/suggest";
import getKind from "../../../lib/getKind";
import {
  Consumer as ItemsConsumer,
  ContextProps
} from "../../../containers/Items";
import Page from "../../templates/CreateScheduleDetail/Page";

export interface State extends ItemDetailParam {
  iconSelected: boolean;
  priority: number;
  suggestList: SuggestItem[];
}

interface Props extends ItemDetailParam {
  navigation: NavigationScreenProp<NavigationRoute>;
}

type PlanProps = Props & Pick<ContextProps, "itemDetails" | "refreshData">;

export default class extends Component<Props> {
  render() {
    return (
      <ItemsConsumer>
        {({ refreshData, itemDetails }: ContextProps) => (
          <Plan
            {...this.props}
            refreshData={refreshData}
            itemDetails={itemDetails}
          />
        )}
      </ItemsConsumer>
    );
  }
}

class Plan extends Component<PlanProps, State> {
  state = {
    title: this.props.title || "",
    kind: this.props.kind || "",
    place: this.props.place || "",
    url: this.props.url || "",
    memo: this.props.memo || "",
    moveMinutes: this.props.moveMinutes || 0,
    iconSelected: false,
    priority: 1,
    suggestList: []
  };

  componentDidMount() {
    const suggestList = (this.props.itemDetails || []).map(itemDetail => ({
      title: itemDetail.title,
      kind: itemDetail.kind
    }));

    this.setState({
      suggestList
    });

    const itemId = this.props.navigation.getParam("itemId", "1");

    db.transaction((tx: SQLite.Transaction) => {
      countByItemId(tx, itemId, this.getCount);
    });
  }

  componentDidUpdate() {
    const kind = this.props.navigation.getParam("kind", "");

    if (!kind) {
      return;
    }

    if (this.state.kind !== kind) {
      this.setState({ kind, iconSelected: true });
    }
  }

  getCount = (count: number, error: ResultError) => {
    if (error) {
      return;
    }

    this.setState({
      priority: count + 1
    });
  };

  onDismiss = () => {
    this.props.navigation.goBack();
  };

  onSave = (
    title: string,
    kind: string,
    place: string,
    url: string,
    memo: string,
    time: number
  ) => {
    const itemId = this.props.navigation.getParam("itemId", "1");

    db.transaction((tx: SQLite.Transaction) => {
      const itemDetail: ItemDetail = {
        itemId,
        title,
        kind,
        place,
        url,
        memo,
        moveMinutes: time,
        priority: this.state.priority
      };

      insertItemDetail(tx, itemDetail, this.save);
    });
  };

  save = (_: number, error: ResultError) => {
    if (error) {
      return;
    }

    const itemId = this.props.navigation.getParam("itemId", "1");

    this.props.navigation.navigate("CreateSchedule", {
      itemId,
      refresh: uuidv1()
    });

    if (this.props.refreshData) {
      this.props.refreshData();
    }
  };

  onIcons = (title: string) => {
    this.props.navigation.navigate("Icons", {
      kind: getKind(title),
      onSelectIcon: (kind: string) => {
        this.props.navigation.navigate("CreateScheduleDetail", {
          kind: kind
        });
      },
      onDismiss: () => {
        this.props.navigation.navigate("CreateScheduleDetail");
      },
      photo: false
    });
  };

  render() {
    return (
      <Page
        title={this.state.title}
        kind={this.state.kind}
        place={this.state.place}
        url={this.state.url}
        memo={this.state.memo}
        time={this.state.moveMinutes}
        suggestList={this.state.suggestList}
        iconSelected={this.state.iconSelected}
        onDismiss={this.onDismiss}
        onSave={this.onSave}
        onIcons={this.onIcons}
      />
    );
  }
}
