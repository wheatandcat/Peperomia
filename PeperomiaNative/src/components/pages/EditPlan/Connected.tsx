import * as SQLite from 'expo-sqlite';
import React, {
  useContext,
  useState,
  memo,
  useEffect,
  useCallback,
} from 'react';
import { NavigationScreenProp, NavigationRoute } from 'react-navigation';
import { Alert } from 'react-native';
import {
  Context as ItemsContext,
  ContextProps as ItemContextProps,
} from '../../../containers/Items';
import { db, ResultError } from '../../../lib/db';
import { update as updateItem, Item } from '../../../lib/db/item';
import {
  update as updateCalendar,
  insert as insertCalendar,
  Calendar,
} from '../../../lib/db/calendar';
import getKind from '../../../lib/getKind';
import { SuggestItem } from '../../../lib/suggest';
import { useDidMount } from '../../../hooks/index';
import Page from '../../templates/CreatePlan/Page';

type Props = {
  navigation: NavigationScreenProp<NavigationRoute>;
};

type ConnectedProps = Pick<
  ItemContextProps,
  'items' | 'refreshData' | 'calendars'
> & {
  navigation: NavigationScreenProp<NavigationRoute>;
};

type PlanProps = Pick<ItemContextProps, 'items' | 'refreshData'> & {
  navigation: NavigationScreenProp<NavigationRoute>;
  input: {
    title: string;
    date: string;
  };
  image: string;
  kind: string;
  onInput: (name: string, value: any) => void;
  onImage: (image: string) => void;
  onSave: () => void;
  onIcons: () => void;
  onHome: () => void;
};

type State = {
  input: {
    title: string;
    date: string;
  };
  image: string;
  kind: string;
  calendar: Calendar;
};

export default (props: Props) => {
  const { refreshData, items, calendars } = useContext(ItemsContext);

  return (
    <Connected
      navigation={props.navigation}
      items={items}
      refreshData={refreshData}
      calendars={calendars}
    />
  );
};

const Connected = memo((props: ConnectedProps) => {
  const [state, setState] = useState<State>({
    input: { title: '', date: '' },
    image: '',
    kind: '',
    calendar: {
      id: 0,
      itemId: 0,
      date: '',
    },
  });

  useDidMount(() => {
    const id = props.navigation.getParam('id', 0);
    let input: {
      date: string;
      title: string;
    } = {
      date: '',
      title: '',
    };

    const calendar = (props.calendars || []).find(
      item => Number(id) === Number(item.itemId)
    );

    if (calendar && calendar.date) {
      input.date = calendar.date;
      setState(s => ({
        ...s,
        ...state,
        calendar,
      }));
    }

    const schedule = (props.items || []).find(
      item => Number(id) === Number(item.id)
    );

    if (schedule && schedule.title) {
      input.title = schedule.title;
    }

    const image = props.navigation.getParam('image', '');
    const kind = props.navigation.getParam('kind', '');

    setState(s => ({
      ...s,
      input,
      image,
      kind,
    }));
  });

  useEffect(() => {
    const image = props.navigation.getParam('image', '');
    if (image && image !== state.image) {
      setState(s => ({
        ...s,
        image,
      }));
    }

    const kind = props.navigation.getParam('kind', '');
    if (kind && kind !== state.kind) {
      setState(s => ({
        ...s,
        kind,
      }));
    }
  }, [props.navigation, state]);

  const onImage = useCallback((image: string) => {
    setState(s => ({
      ...s,
      image,
    }));
  }, []);

  const onInput = useCallback(
    (name: string, value: any) => {
      const input = {
        ...state.input,
        [name]: value,
      };

      setState(s => ({
        ...s,
        input,
      }));
    },
    [state]
  );

  const save = useCallback(
    (_: Item[], error: ResultError) => {
      if (error) {
        Alert.alert('保存に失敗しました');
        return;
      }

      const id = props.navigation.getParam('id', 0);

      props.navigation.navigate('Schedule', {
        itemId: id,
        title: state.input.title,
      });
    },
    [props.navigation, state.input.title]
  );

  const saveCalendar = useCallback(
    (_: Calendar | number, error: ResultError) => {
      if (error) {
        return;
      }

      if (props.refreshData) {
        props.refreshData();
      }
    },
    [props]
  );

  const onSave = useCallback(async () => {
    if (state.input.date) {
      const check = (props.calendars || []).find(
        calendar => calendar.date === state.input.date
      );

      if (check) {
        Alert.alert('同じ日にスケジュールが既に登録されています');
        return;
      }
    }

    db.transaction((tx: SQLite.Transaction) => {
      const id = props.navigation.getParam('id', 0);

      const item: Item = {
        id,
        title: state.input.title,
        kind: state.kind || getKind(state.input.title),
        image: '',
      };

      updateItem(tx, item, save);

      if (!state.input.date) {
        return;
      }

      if (state.calendar.id) {
        updateCalendar(
          tx,
          {
            ...state.calendar,
            date: state.input.date,
          },
          saveCalendar
        );
      } else {
        insertCalendar(
          tx,
          {
            itemId: id,
            date: state.input.date,
          },
          saveCalendar
        );
      }
    });
  }, [
    props.calendars,
    props.navigation,
    save,
    saveCalendar,
    state.calendar,
    state.input.date,
    state.input.title,
    state.kind,
  ]);

  const onIcons = useCallback(() => {
    props.navigation.navigate('Icons', {
      kind: getKind(state.input.title),
      onSelectIcon: (kind: string) => {
        props.navigation.navigate('EditPlan', {
          kind: kind,
        });
      },
      onDismiss: () => {
        props.navigation.navigate('EditPlan');
      },
      photo: true,
    });
  }, [props.navigation, state.input.title]);

  const onHome = useCallback(() => {
    props.navigation.goBack();
  }, [props.navigation]);

  console.log(state);

  return (
    <Plan
      navigation={props.navigation}
      input={state.input}
      image={state.image}
      kind={state.kind}
      items={props.items}
      refreshData={props.refreshData}
      onInput={onInput}
      onImage={onImage}
      onSave={onSave}
      onIcons={onIcons}
      onHome={onHome}
    />
  );
});

type PlanState = {
  suggestList: SuggestItem[];
};

const Plan = memo((props: PlanProps) => {
  const [state, setState] = useState<PlanState>({
    suggestList: [],
  });

  useDidMount(() => {
    const suggestList = (props.items || []).map(item => ({
      title: item.title,
      kind: item.kind,
    }));

    setState(s => ({
      ...s,
      suggestList,
    }));
  });

  const onSave = useCallback(async () => {
    await props.onSave();

    if (props.refreshData) {
      props.refreshData();
    }
  }, [props]);

  return (
    <Page
      mode="edit"
      title={props.input.title}
      date={props.input.date}
      image={props.image}
      kind={props.kind}
      suggestList={state.suggestList}
      onInput={props.onInput}
      onImage={props.onImage}
      onSave={onSave}
      onIcons={props.onIcons}
      onHome={props.onHome}
    />
  );
});
