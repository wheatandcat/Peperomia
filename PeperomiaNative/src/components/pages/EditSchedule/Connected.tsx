import { SQLite } from "expo-sqlite";
import React, { Component } from "react";
import { NavigationScreenProp, NavigationRoute } from "react-navigation";
import uuidv1 from "uuid/v1";
import { db } from "../../../lib/db";
import { selectByItemId } from "../../../lib/db/itemDetail";
import { ItemProps } from "../../organisms/Schedule/Cards";
import Page from "../../templates/CreateSchedule/Page";

interface Props {
  title: string;
  items: ItemProps[];
  navigation: NavigationScreenProp<NavigationRoute>;
  onShow: () => void;
}

export interface State {
  items: ItemProps[];
  refresh: string;
}

export default class extends Component<Props, State> {
  state = { items: this.props.items || [], refresh: "0" };

  componentDidUpdate() {
    const refresh = this.props.navigation.getParam("refresh", "0");

    if (refresh !== this.state.refresh) {
      const itemId = this.props.navigation.getParam("itemId", "1");
      db.transaction((tx: SQLite.Transaction) => {
        selectByItemId(tx, itemId, this.setItems);
      });
      this.setState({ refresh });
    }
  }

  setParams = (data: any, error: any) => {
    if (error) {
      return;
    }

    const itemId = this.props.navigation.getParam("itemId", "1");
    const mode = this.props.navigation.getParam("mode", "create");

    this.props.navigation.setParams({
      title: data.title,
      itemId,
      mode
    });
  };

  setItems = (data: any, error: any) => {
    if (error) {
      return;
    }

    this.setState({ items: data });
  };

  onCreateScheduleDetail = () => {
    const itemId = this.props.navigation.getParam("itemId", "1");
    this.props.navigation.navigate("CreateScheduleDetail", {
      itemId,
      onSave: () => {
        this.props.navigation.navigate("CreateSchedule", {
          itemId: itemId,
          refresh: uuidv1()
        });
      }
    });
  };

  onScheduleDetail = (id: string) => {
    this.props.navigation.navigate("ScheduleDetail", {
      scheduleDetailId: id,
      priority: this.state.items.length + 1
    });
  };

  render() {
    return (
      <Page
        data={this.state.items}
        onScheduleDetail={this.onScheduleDetail}
        onCreateScheduleDetail={this.onCreateScheduleDetail}
      />
    );
  }
}
