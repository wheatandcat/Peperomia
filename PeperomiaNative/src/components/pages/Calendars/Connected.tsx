import React, { Component } from 'react';
import { NavigationScreenProp, NavigationRoute } from 'react-navigation';
import { createStackNavigator } from 'react-navigation-stack';
import {
  Consumer as ItemsConsumer,
  ContextProps,
} from '../../../containers/Items';
import theme from '../../../config/theme';
import Schedule from '../Schedule/Switch';
import EditPlan from '../EditPlan/Connected';
import Page from './Page';

type Props = {
  navigation: NavigationScreenProp<NavigationRoute>;
};

class Container extends Component<Props> {
  static navigationOptions = () => {
    return {
      header: null,
    };
  };

  onCreate = (date: string) => {
    this.props.navigation.navigate('CreatePlan', {
      date,
    });
  };

  onSchedule = (id: number, title: string) => {
    this.props.navigation.navigate('Schedule', { itemId: id, title });
  };

  render() {
    return (
      <ItemsConsumer>
        {({ calendars, itemsLoading }: ContextProps) => (
          <Page
            calendars={calendars || []}
            loading={itemsLoading || false}
            onCreate={this.onCreate}
            onSchedule={this.onSchedule}
          />
        )}
      </ItemsConsumer>
    );
  }
}

const MainCardNavigator = createStackNavigator(
  {
    Container: {
      screen: Container,
    },
    Schedule: {
      screen: Schedule,
    },
  },
  {
    defaultNavigationOptions: {
      headerStyle: {
        backgroundColor: theme().mode.header.backgroundColor,
      },
      headerTitleStyle: {
        color: theme().mode.header.text,
      },
      headerTintColor: theme().mode.header.text,
    },
  }
);

export default createStackNavigator(
  {
    MainCardNavigator: {
      screen: MainCardNavigator,
    },
    EditPlan: {
      screen: EditPlan,
    },
  },
  {
    initialRouteName: 'MainCardNavigator',
    headerMode: 'none',
  }
);
