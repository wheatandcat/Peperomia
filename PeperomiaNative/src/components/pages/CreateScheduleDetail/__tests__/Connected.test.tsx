import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { itemDetailsMockData } from '__mockData__/itemDetail.ts';
import Connected from '../Connected';

describe('components/pages/CreateScheduleDetail/Connected.tsx', () => {
  let wrapper: ShallowWrapper;

  const propsData: any = () => ({
    navigation: {
      setParams: jest.fn(),
      navigate: jest.fn(),
    },
    route: {
      params: {
        itemId: '1',
        refresh: '',
      },
    },
    refreshData: jest.fn(),
    itemDetails: itemDetailsMockData,
    uid: 'test',
  });

  beforeEach(() => {
    wrapper = shallow(<Connected {...propsData()} />);
  });

  it('正常に表示されている', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
