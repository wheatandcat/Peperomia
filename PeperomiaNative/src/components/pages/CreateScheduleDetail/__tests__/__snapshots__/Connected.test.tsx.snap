// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/CreateScheduleDetail/Connected.tsx 正常に表示されている 1`] = `
<ConnectedActionSheet
  iconSelected={false}
  kind=""
  memo=""
  moveMinutes={0}
  onDismiss={[Function]}
  onIcons={[Function]}
  onSave={[Function]}
  place=""
  suggestList={Array []}
  title=""
  url=""
/>
`;
